File: 백준/Silver/11723. 집합/집합.cc
#include <stdio.h>
#include <string.h>

int main() {
    int M, x, S = 0;
    char cmd[10];

    scanf("%d", &M);

    while (M--) {
        scanf("%s", cmd);

        if (!strcmp(cmd, "add")) {
            scanf("%d", &x);
            S |= (1 << x);
        }
        else if (!strcmp(cmd, "remove")) {
            scanf("%d", &x);
            S &= ~(1 << x);
        }
        else if (!strcmp(cmd, "check")) {
            scanf("%d", &x);
            printf("%d\n", (S & (1 << x)) ? 1 : 0);
        }
        else if (!strcmp(cmd, "toggle")) {
            scanf("%d", &x);
            S ^= (1 << x);
        }
        else if (!strcmp(cmd, "all")) {
            S = (1 << 21) - 1;
        }
        else if (!strcmp(cmd, "empty")) {
            S = 0;
        }
    }

    return 0;
}

File: 백준/Silver/11723. 집합/집합.py
import sys
input = sys.stdin.readline

S = 0  # 집합을 비트로 표현

for _ in range(int(input())):
    cmd = input().strip().split()

    if cmd[0] == 'add':
        S |= (1 << int(cmd[1]))
    elif cmd[0] == 'remove':
        S &= ~(1 << int(cmd[1]))
    elif cmd[0] == 'check':
        print(1 if S & (1 << int(cmd[1])) else 0)
    elif cmd[0] == 'toggle':
        S ^= (1 << int(cmd[1]))
    elif cmd[0] == 'all':
        S = (1 << 21) - 1  # 0 ~ 20번 비트 전부 1
    elif cmd[0] == 'empty':
        S = 0
